src/sinks/pulsar.rs

```diff
@@ -182,7 +221,7 @@ impl Sink<Event> for PulsarSink {
             "Expected `poll_ready` to be called first."
         );
 
-        let message = encode_event(item, &self.encoding).map_err(|_| ())?;
+        let message = encode_event(item, &self.encoding, &self.avro_schema).map_err(|e| error!("Error during encoding: {}", e))?;
```

Comment:
Ah, sorry, this should be an internal event. Can you create a `pulsar.rs` in https://github.com/timberio/vector/tree/master/src/internal_events and create an error like: https://github.com/timberio/vector/blob/e769bccd78bbfa3098282b85567dad0e4f72763e/src/internal_events/remove_fields.rs#L13-L22 ?

Then, you can call it like so:

https://github.com/timberio/vector/blob/e769bccd78bbfa3098282b85567dad0e4f72763e/src/transforms/remove_fields.rs#L72-L74